File: code_samples/scala/algorithms_sorting.scala
/**
 * Algorithms: Sorting
 * AI/ML Training Sample
 */

class Sorting {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object SortingExample {
  def main(args: Array[String]): Unit = {
    val instance = new Sorting()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/go/algorithms_sorting.go
package main

import (
    "fmt"
)

// Algorithms: Sorting
// AI/ML Training Sample

type Sorting struct {
    Data string
}

func NewSorting() *Sorting {
    return &Sorting{
        Data: "",
    }
}

func (s *Sorting) Process(input string) {
    s.Data = input
}

func (s *Sorting) Validate() bool {
    return len(s.Data) > 0
}

func (s *Sorting) GetData() string {
    return s.Data
}

func main() {
    instance := NewSorting()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
